Pre-Generate Docs, Add Support for Plugin-Provided Docs (#1601)
nautobot/nautobot: nautobot/docs/plugins/development.md

| `config_view_name` | `None` | [URL name](#adding-links-to-the-installed-plugins-view) for a "configuration" view defined by this plugin |
| `default_settings` | `{}` | A dictionary of configuration parameters and their default values |
| `home_view_name` | `None` | [URL name](#adding-links-to-the-installed-plugins-view) for a "home" or "dashboard" view defined by this plugin |

> | `docs_view_name` | `None` | [URL name](#adding-links-to-the-installed-plugins-view) for a docs view defined by this plugin |

Reviewer: ```suggestion
| `docs_view_name` | `None` | [URL name](#adding-links-to-the-installed-plugins-view) for a "documentation" view defined by this plugin |
```

!!! warning
    This example is provided as a minimal reference implementation only. It does not address authentication, performance, or the myriad of other concerns that plugin authors should have.

## Adding Help Documentation

> If you are using the `generic.ObjectEditView` from Nautobot for your object, the form will include a help icon with a link to that object's documentation. In order to resolve this link, the view expects to find the documentation for this object in a specific directory tree:

Reviewer: If we can make the functional changes suggested in my other comments, perhaps:

```suggestion
If you are using the `generic.ObjectEditView` from Nautobot for your object, the form can automatically include a help icon with a link to that object's documentation. For this to happen, Nautobot must be able to find the documentation for this object in a specific directory tree within your plugin:
```

> ```no-highlight
> plugin_name/
>   - static/
>     - plugin_name/
>       - docs/
>         - index.html
>         - models/
>           - object_model.html
> ```

Reviewer: Perhaps:

````suggestion
```no-highlight
plugin_name/                   # "nautobot_animal_sounds"
  - static/
    - plugin_name/             # "nautobot_animal_sounds"
      - docs/
        - index.html
        - models/
          - object_model.html  # "animal.html"
```
````

!!! warning
    This example is provided as a minimal reference implementation only. It does not address authentication, performance, or the myriad of other concerns that plugin authors should have.

## Adding Help Documentation

If you are using the `generic.ObjectEditView` from Nautobot for your object, the form will include a help icon with a link to that object's documentation. In order to resolve this link, the view expects to find the documentation for this object in a specific directory tree:

```no-highlight

> plugin_name/
>   - plugin_name/
>     - static/
>       - plugin_name/
>         - docs/
>           - index.html
>           - models/
>             - object_model.html

Reviewer: I don't think the first `plugin_name` is actually needed - it's just the folder that contains the plugin root directory, and isn't actually part of the loading path.

```suggestion
plugin_name/
  - static/
    - plugin_name/
      - docs/
        - index.html
        - models/
          - object_model.html
```
